### ISSUE TITLE
performance of .I when subsetting rows from a global identifier

### ISSUE THREAD

**Issue by aksa111** [Rdatatable/data.table#3622]
This is related to the SO quesiton https://stackoverflow.com/questions/56408936/performance-issue-with-sd-when-indexing-by-variable/56409174#

`set.seed(102938)

dd <- data.table(id = rep(1:10000, each = 10), seq = seq(1:10))
setkey(dd, id)

N <- 2
microbenchmark(dd[,.SD[2], keyby = id],
               dd[,.SD[N], keyby = id],
               times = 5)`

The suggested approach was to use `.I`, though it improved the performance, it didn't match the performance of inserting the index value directly.  Is it the search time for the identifier in the global env.

`microbenchmark(dd[,.SD[2], keyby = id],
                dd[dd[,.I[N], keyby = id]$V1],
                times = 5)`

**Comment by MichaelChirico** [comment]
can you add the output of setting verbose = TRUE to each of the lines?

**Comment by aksa111** [comment]
Thank you.  I think the GForce optimization is the key

**Comment by MichaelChirico** [comment]
yes, that's what I suspected. still, we may be able to turn on GForce in this situation, though I'm not sure it will be possible to do so efficiently. something to explore.

**Comment by ben-schwen** [comment]
Actually lapply optimization is not on in this case, so GForce can't turn on.
But even for the simpler example GForce does also not kick in.

```r
dd <- data.table(id = rep(1:10000, each = 10), seq = seq(1:10))
N <- 2
dd[,seq[N], keyby = id]
```

We could ofc check in argument `is.symbol` and evaluate like we do for other arguments but it feels like the whole `.gforce_ok` function needs a proper rewrite.
